src/Simulation/DocumentationGenerationStep/ProcessDocComments.cs

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Quantum.QsCompiler;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.Documentation;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.Core;
+using Newtonsoft.Json.Linq;
+
+#nullable enable
+
+namespace Microsoft.Quantum.Documentation
+{
+    public class ProcessDocComments
+    : SyntaxTreeTransformation<ProcessDocComments.TransformationState>
+    {
+        public class TransformationState
+        { }
+
+        private string? OutputPath;
+
+        public ProcessDocComments(
+            string? outputPath = null
+        )
+        : base(new TransformationState())
+        {
+            OutputPath = outputPath;
+            // If the output path is not null, make sure the directory exists.
+            if (outputPath != null && !Directory.Exists(outputPath))
+            {
+                Directory.CreateDirectory(outputPath);
+            }
+
+            // We provide our own custom namespace transformation, and expression kind transformation.
+            this.Namespaces = new ProcessDocComments.NamespaceTransformation(this, outputPath);
```

Comment:
I'd suggest to disable transformations on nodes that are not needed. This can be done by setting 
`this.Statements = new StatementTransformation(this, TransformationOptions.Disabled);
this.Expressions = ...; 
this.Types = ...`
Since you only need the top-level transformation, you alternatively just pass TransformationOptions.Disabled to the base constructor, and setting this.Namespaces here makes sure that the transformation on the namespace executed (though this.Apply would be disabled as well).